### PR TITLE
Fetch-values for implicit actions save QueryExecution in action context

### PR DESCRIPTION
**Pull request by qnkhuat**
Previously fetching-values for an action save a QueryExecution with `:context :question` which doesn't make sense. 
This PR changes it so that we save it as a new context name `:action`

To test:

0. do `(dev/track! :model/QueryExecution)`
1. Create a simple model
2. Enable implicit actions
3. Create a dashboard with an ID filter
4. Add an action link to the Update action added above, Select ID field and map to ID filter
6. save the dashboard
7. Enter an id for the ID filter and click onto the update action
8. You should see a log for insert into `QueryExecution` table with `{:context :action, :action_id ...}`

### CODE REVIEW

File: test/metabase/actions/execution_test.clj [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    (mt/with-actions-enabled
+      (mt/with-actions [_                   {:dataset true :dataset_query (mt/mbql-query venues {:fields [$id $name]})}
+                        {:keys [action-id]} {:type :implicit :kind "row/update"}]
+        (is (= {"id" 1 "name" "Red Medicine"}
+               (actions.execution/fetch-values (action/select-action :id action-id) {"id" 1})))
+
+        (t2/select :model/QueryExecution :context :action)
+        ;; the query execution is saved async, so we need to sleep a bit
+        (Thread/sleep 200)
+        (is (true? (t2/exists? :model/QueryExecution :action_id action-id)))))))
```
Comment (calherries) [inline test/metabase/actions/execution_test.clj:22]: We should test that `context=action` here
```suggestion
        (is (true? (t2/exists? :model/QueryExecution :action_id action-id :context :action)))))))
```

**Review by calherries — CHANGES_REQUESTED** [review]

File: test/metabase/actions/execution_test.clj [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+(deftest fetch-values-save-execution-info-test
+  (testing "fetch values for implicit action will save an execution info"
+    (mt/with-actions-enabled
+      (mt/with-actions [_                   {:dataset true :dataset_query (mt/mbql-query venues {:fields [$id $name]})}
+                        {:keys [action-id]} {:type :implicit :kind "row/update"}]
+        (is (= {"id" 1 "name" "Red Medicine"}
+               (actions.execution/fetch-values (action/select-action :id action-id) {"id" 1})))
+
+        (t2/select :model/QueryExecution :context :action)
```
[the remark this answers is not in the document]
Comment (qnkhuat) [reply test/metabase/actions/execution_test.clj]: oops. Removed it in 352dbc6793

File: src/metabase/api/dashboard.clj [hunk 3]
```diff
@@ -955,15 +956,24 @@
 
 
 ;;; ---------------------------------- Executing the action associated with a Dashcard -------------------------------
-#_{:clj-kondo/ignore [:deprecated-var]}
-(api/defendpoint-schema GET "/:dashboard-id/dashcard/:dashcard-id/execute"
+
+(defn action-for-dashcard-id
+  "Given a `dashcard-id`, returns the action associated with it if any."
+  [dashcard-id]
+  (->> (t2/select-one-fn :action_id :model/DashboardCard
```
[the remark this answers is not in the document]
Comment (qnkhuat) [reply src/metabase/api/dashboard.clj]: good point. I moved it into `dashboard-card` in 0adf723815

**Review by calherries — APPROVED** [review]
LGTM

**Pull request merged**